rust/foxglove/src/websocket_server.rs

```diff
@@ -168,6 +168,22 @@ impl WebSocketServerHandle {
             .await;
     }
 
+    /// Publishes a status message to all clients.
+    pub fn publish_status(
+        &self,
+        level: crate::websocket::StatusLevel,
+        message: impl Into<String>,
+        id: Option<impl Into<String>>,
+    ) {
+        self.0
+            .publish_status(level, message.into(), id.map(|id| id.into()));
+    }
+
+    /// Removes status messages by id from all clients.
+    pub fn remove_status(&self, status_ids: Vec<String>) {
```

Comment:
Maybe make it a bit more generic?

```suggestion
    pub fn remove_status<II, S>(&self, status_ids: II)
    where
        II: IntoIterator<Item = S>,
        S: Into<String>,
    {
```

Or maybe that's overkill.

Reply:
I did that initially, but we end up needing a Vec<String> anyway, and maybe I'm wrong but I think IntoIterator<String>.collect::<Vec<String>>() will create a new Vect, so then it's just worse in the common case where you already had a Vec.

Better to get the user to do the collect into a Vec if they have an iterator, or just pass it in if they have a Vec.